src/main/java/bio/terra/service/snapshot/SnapshotService.java

```diff
@@ -177,6 +191,54 @@ public String exportSnapshot(
         .submit();
   }
 
+  /**
+   * @param userReq authenticated user
+   * @return accessible snapshot UUIDs and the IamRoles dictating their accessibility, established
+   *     directly by SAM and indirectly by any linked RAS passport
+   */
+  public Map<UUID, Set<IamRole>> listAuthorizedSnapshots(AuthenticatedUserRequest userReq) {
+    return combineIdsAndRoles(
+        iamService.listAuthorizedResources(userReq, IamResourceType.DATASNAPSHOT),
+        listRasAuthorizedSnapshots(userReq));
+  }
+
+  /**
+   * @param userReq authenticated user
+   * @return accessible snapshot UUIDs and the IamRoles dictating their accessibility, established
+   *     indirectly by any linked RAS passport
+   */
+  public Map<UUID, Set<IamRole>> listRasAuthorizedSnapshots(AuthenticatedUserRequest userReq) {
+    Map<UUID, Set<IamRole>> idsAndRoles = new HashMap<>();
+    try {
+      List<RasDbgapPermissions> permissions = ecmService.getRasDbgapPermissions(userReq);
+      if (!permissions.isEmpty()) {
+        List<UUID> uuids = snapshotDao.getAccessibleSnapshots(permissions);
+        Set<IamRole> roles = Set.of(IamRole.READER);
+        uuids.forEach(uuid -> idsAndRoles.put(uuid, roles));
+      }
+    } catch (ParseException ex) {
+      // Not sure if we want to catch any exception, or catch upstream instead.
```

Comment:
I raised the log level to `error`, but agree that we wouldn't have a reason to look at these logs proactively.

@nmalfroy any thoughts on this?

I am inclined to leave the current treatment and ticket any concerns separately, since passports are not live in production.

Two alternatives that come to mind:

- Expand EnumerateSnapshotModel to include error information.  Would benefit from more discussion.
- @samanehsan originally brought up in the Spike document the idea of "includ[ing] a new parameter option `includePassportAuth` that is false by default".  Then instead of swallowing the exception we can return it, and the caller has a workaround available of calling the endpoint without passport authorizations checked.